solr/core/src/java/org/apache/solr/cluster/placement/impl/AttributeFetcherImpl.java

```diff
@@ -197,32 +148,78 @@ public AttributeValues fetchAttributes() {
       }
     }
 
-    return new AttributeValuesImpl(nodeToCoreCount,
-        nodeToDiskType,
-        nodeToFreeDisk,
-        nodeToTotalDisk,
-        nodeToHeapUsage,
-        nodeToSystemLoadAverage,
-        syspropSnitchToNodeToValue,
-        metricSnitchToNodeToValue);
+    for (Node node : nodeToReplicaInternalTags.keySet()) {
```

Comment:
I understand here we might be contacting nodes again to fetch replica metrics. I think this is ok but maybe add a comment that there can be an optimization where node metrics and replica metrics are fetched with a single call to each node?
Here we might even contact each node multiple times I believe depending on the collections we need info about and how their replicas are distributed?

Reply:
Yes, we make two calls per node - one is `getNodeValues` and the other is `getReplicaInfos`.

The number of calls doesn't depend on the number of collections - we already use a set of nodes that covers all replicas for all interesting collections and we only call `getReplicaInfo` once per node (and it returns all replica infos, regardless of collection).